src/Castle.Core/Core/Internal/ILockHolder.cs

```diff
@@ -15,7 +15,10 @@
 namespace Castle.Core.Internal
 {
 	using System;
+	using System.ComponentModel;
 
+	[Obsolete("Consider using `System.Threading.ReaderWriterLockSlim` instead of `Lock` and related types.")] // TODO: Remove this type.
+	[EditorBrowsable(EditorBrowsableState.Never)]
```

Comment:
I've put `// TODO` comments next to the most `[Obsolete]` attributes. They contain instructions how to deal with each obsolete member, so that the clean-up will be as simple as possible.